controllers/userController.js

```diff
@@ -1,90 +1,348 @@
+    return res.send({status: STATUS_CODE.INVALID_CODE});
+  }
+  //check for user status 
+  const queryEmail = `SELECT * FROM users WHERE email = ?`;
+  connection.promise().query(queryEmail, [email])
   .then(data => {
-    if (data[0].affectedRows) {
-      message = 'user deleted successfully';
-      res.status(200).send({message: message, data: data[0]});
+    if (data[0].length === 0) {
+      console.log(data[0])
+      return res.send({status: STATUS_CODE.INVALID_EMAIL});
+    } else {
+      //data[0][0] contains the actual json object that gets returned by mysql
+      if (data[0][0].status !== REGISTRATION_STATUS.UNREGISTERED){
+        return res.send({status : STATUS_CODE.USER_EXISTENT});
+      }
     }
   })
   .catch(error => {
-    res.status(400).send({message: message, error: error});
+    console.log(error);
+    return res.send({ status: STATUS_CODE.ERROR});
   });
-
-  return res;
-};
-
-//edit one user
-exports.edituser = async (req, res) => {
-  const { email, name, type, visions} = req.body;
-  let message = 'Error in editing user';
+  // Creating a unique salt for a particular user 
+  const salt = crypto.randomBytes(16).toString('hex'); 
   const query = `UPDATE users 
-  SET name = ?, type = ?, visions = ?
+  SET hash = ?, salt = ?, status = 'registered'
   WHERE email = ?`;
+  const hash = hashPassword(password, salt);
 
-  connection.promise().query(query, [name, type, visions, email])
+  connection.promise().query(query, [hash, salt, email])
   .then(data => {
     if (data[0].affectedRows) {
-      message = 'user edited successfully';
-      console.log(message);
-      res.status(200).send({message: message, data: data[0]});
+      return res.send({status : STATUS_CODE.SUCCESS});
     }
   })
   .catch(error => {
-    res.status(400).send({message: message, error: error});
+    console.log(error);
+    return res.send({ status: STATUS_CODE.ERROR});
   });
-};
\ No newline at end of file
+}
+
+function hashPassword (password, salt) {
+   // Hashing user's salt and password with 1000 iterations, 
+   hash = crypto.pbkdf2Sync(password, salt, 1000, 64, `sha512`).toString(`hex`); 
+   return hash;
+}
+
+//edit one user
+exports.updateUser = async (req, res) => {
+  const { old_email, email, name, type, visions } = req.body;
+
+  try{
+    let verify = await this.verifyUser( old_email );
+    
+    console.log(verify);
+
+    if (verify.NUM == 0) {
+      return res.send({ status: STATUS_CODE.INCORRECT_USER_EMAIL, result: old_email });
+    }
+
+    let remove = await this.removeUser( old_email );
+    if (remove.affectedRows){
+      console.log('update in process....');
+    }
+
+    let result = await this.insertUser({ email, name, type, visions });
+    if (result.affectedRows) {
+      return res.send({ status: STATUS_CODE.SUCCESS });
+    }
+  } catch (error){
+    return res.send({ status: STATUS_CODE.ERROR, result: error });
+  }
+
+  return res.send({ status: STATUS_CODE.ERROR });
```

Comment:
I think we can reformat the code to have all the APIs for user log in and registration in another controller, but I think this is just a minor suggestion to consider if we have time later on

Reply:
I agree!!!